client/blocks/site-address-changer/index.jsx

```diff
@@ -383,15 +391,25 @@ export class SiteAddressChanger extends Component {
 				</FormSectionHeading>
 				<div className="site-address-changer__info">
 					<p>
-						{ translate(
-							'Once you change your site address, %(currentDomainName)s will no longer be available. {{a}}Did you want to add a custom domain instead?{{/a}}',
-							{
-								args: { currentDomainName },
-								components: {
-									a: <a href={ addDomainPath } onClick={ this.handleAddDomainClick } />,
-								},
-							}
-						) }
+						{ hasNonWpcomDomains &&
+						i18n.hasTranslation(
+							'Once you change your site address, %(currentDomainName)s will no longer be available.'
+						)
+							? translate(
+									'Once you change your site address, %(currentDomainName)s will no longer be available.',
+									{
+										args: { currentDomainName },
+									}
+							  )
+							: translate(
+									'Once you change your site address, %(currentDomainName)s will no longer be available. {{a}}Did you want to add a custom domain instead?{{/a}}',
+									{
+										components: {
+											args: { currentDomainName },
```

Comment:
Needs moving up a level I think (args is not a component)

![Screenshot 2023-10-27 at 15-33-35 Test Business Simple Site — WordPress com](https://github.com/Automattic/wp-calypso/assets/811776/139f5ddd-289d-4a8e-a2a6-0ac608dbd87d)

Reply:
Feel free to simplify and drop the hasTranslation check if that's easier too.

Reply:
Updated and dropped hasTranslation